File: python/program/Gpu.py
from Komponen import Komponen
from Memory import Memory


class Gpu(Komponen, Memory):
    def __init__(self, gddr, bandwidth, kapasitasGB, kecepatanMHz, merk, nama):
        Komponen.__init__(self, merk, nama)
        Memory.__init__(self, kapasitasGB, kecepatanMHz)
        self.gddr = gddr
        self.bandwidth = bandwidth

    def set_gddr(self, gddr):
        self.gddr = gddr

    def set_bandwidth(self, bandwidth):
        self.bandwidth = bandwidth

    def get_gddr(self):
        return self.gddr

    def get_bandwidth(self):
        return self.bandwidth

File: java/program/Gpu.java
class Gpu extends Komponen {
    private String gddr;
    private String bandwidth;
    private Memory memory;

    public Gpu() {
        memory = new Memory();
    }

    public Gpu(String gddr, String bandwidth, int kapasitasGB, int kecepatanMHz, String merk, String nama) {
        super(merk, nama);
        memory = new Memory(kapasitasGB, kecepatanMHz);
        this.gddr = gddr;
        this.bandwidth = bandwidth;
    }

    public void setGddr(String gddr) {
        this.gddr = gddr;
    }

    public void setBandwidth(String bandwidth) {
        this.bandwidth = bandwidth;
    }

    public String getGddr() {
        return this.gddr;
    }

    public String getBandwidth() {
        return this.bandwidth;
    }

    // Memory methods
    public void setKapasitasGB(int kapasitasGB) {
        this.memory.setKapasitasGB(kapasitasGB);
    }

    public void setKecepatanMHz(int kecepatanMHz) {
        this.memory.setKecepatanMHz(kecepatanMHz);
    }

    public int getKapasitasGB() {
        return this.memory.getKapasitasGB();
    }

    public int getKecepatanMHz() {
        return this.memory.getKecepatanMHz();
    }
}
